### DISCUSSION TITLE
Using Longhorn on multiple NFS shares

### DISCUSSION THREAD

**Discussion by urbaman** [longhorn/longhorn#5309] [Q&A] [answered]
Hi,

I'm planning the architecture of my future K8s cluster, and I would like to know if and how I could use Longhorn as a storage manager in the following stack:

3x K8s contron planes
3x k8s workers

1x HDD NFS share (for documents, probably through nextcloud)
1x SDD big share (for media files, probably through plex)
1x SDD small share (for actual k8s pods outside those above)

I do not need redundancy probably, as I'm already doing it underneath the NFS shares.

So, I'd give the three shares to all of the k8s' nodes, and would like to use Longhorn as the storage/volume orchestrator the the cluster.

Will it work? Does it presumes different disks/shares to different nodes?

I could also divide the shares between all of the nodes, without redundancy to keep space. Is this better than using the same shares for all of the nodes?

Thank you very much.

**Comment by joshimoo** [comment, marked as answer]
A mounted NFS share will not work for the underlying storage of longhorn replicas.
The replica storage would require a ext4/xfs filesystem preferably on a local ssd drive.

**Comment by urbaman** [comment]
So, I can use external storage only in the form of block devices, right?

Thank you.
